src/Ads/AccountService.php

```diff
@@ -252,17 +252,18 @@ private function check_billing_status( array $account ) {
 	/**
 	 * Get the callback function for linking a merchant account.
 	 *
-	 * @throws Exception When the merchant or ads account hasn't been set yet.
+	 * @throws Exception When the ads account hasn't been set yet.
 	 */
 	private function link_merchant_account() {
-		if ( ! $this->options->get_merchant_id() ) {
-			throw new Exception( 'A Merchant Center account must be connected' );
-		}
-
 		if ( ! $this->options->get_ads_id() ) {
 			throw new Exception( 'An Ads account must be connected' );
 		}
 
+		if ( ! $this->options->get_merchant_id() ) {
+			// Return early if the merchant account hasn't been created yet.
+			return;
```

Comment:
If we return early here then the AdsAccountState will mark the step `link_merchant` as completed. I see that the MerchantAccountState does the opposite of this for the new step `link_ads`. And the only thing they rely on is the ID of the opposite account to allow the linking to happen.

Maybe it's a little early and out of scope for this PR but eventually we should only link if the state of the other account has reached a certain step, for example step `link_ads` can only be completed if ads steps `set_id` and `access_account` have been completed.

So this makes me wonder if we should make this part a bit more flexible and define a fourth state:
```php
	public const STEP_SKIPPED = 2;
```

We'll need to change several comparisons from `$step === STEP_DONE` to `$step >= STEP_DONE` but it would give us some more flexibility into knowing which steps were completed and which ones were skipped/optional.

Reply:
@mikkamp I think as of now we can skip adding a new flag

```php
public const STEP_SKIPPED = 2;
```

Introducing the new flag would require some checks to perform at several places and also make changes to several places including unit tests, it will also open some possibility to introduce any possible issue.

Instead, we can call `link_merchant_account` conditionally if the `ads_id` and `merchant_id` is present. This change is introduced for both AccountService.php for `Ads` and `Merchant` classes.

Reply:
Thanks for clarifying, only working with the available STEP statuses sounds like a good approach.